mysql-test/t/ps_trigger_access_check_regression.test

```diff
@@ -0,0 +1,52 @@
+#
+# PS-7778  Wrongly handled definer for triggers when prepared statements are used
+#
+
+--echo # Create users 'user01' and 'user02'.
+CREATE USER 'user01'@'%' IDENTIFIED BY 'user01';
+CREATE USER 'user02'@'%' IDENTIFIED BY 'user02';
+GRANT ALL PRIVILEGES ON *.* to 'user01'@'%';
+
+--echo #
+--echo # Fill tables.
+CREATE DATABASE test01;
+use test01;
+CREATE TABLE c (id int primary key, a varchar(100));
+INSERT INTO c SET id =1, a=now();
+use test;
+CREATE TABLE tbl01 (id int primary key, a varchar(100));
+CREATE TABLE tbl02 (id int primary key, a varchar(100));
+GRANT SELECT, UPDATE, INSERT on test.tbl01 TO 'user02'@'%';
+INSERT INTO tbl01 SET id = 1, a=now();
+
+--echo #
+--echo # Create the trigger.
+CREATE DEFINER='user01'@'%' TRIGGER trg01 AFTER UPDATE ON tbl01 FOR EACH ROW UPDATE tbl02 SET a = (SELECT a FROM test01.c WHERE id =1) WHERE id=NEW.id;
+
+
+--echo #
+--echo # Log in as 'user02'.
+--connect (con_user02, localhost, user02, user02, )
+--connection con_user02
+
+use test;
+
+--echo #
+--echo # Update should succeed.
+update tbl01 set a = now();
```

Comment:
s/update/UPDATE
s/set/SET

Reply:
I've rewritten the test, and hopefully have all keywords in upper case now.